### ISSUE TITLE
no-return-assignment ignore arrow function

### ISSUE THREAD

**Issue by ZimGil** [eslint/eslint#13047]
**What rule do you want to change?**
`no-return-assignment

**Does this change cause the rule to produce more or fewer warnings?**
Same or fewer depends on user configurations

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option

**Please provide some example code that this change will affect:**
Settings the option to something like: `allowArrowFunctions: true`
```js
// This will not throw an error
(foo) => this.foo = foo

// This will throw an error
(foo) => {
  return this.foo = foo;
}
```

**What does the rule currently do for this code?**
The rule will throw an error for any assignment in return statement

**What will the rule do after it's changed?**
The rule will have an option to ignore arrow function single line return.

**Are you willing to submit a pull request to implement this change?**
I would but I currently don't have the time.

**Comment by anikethsaha** [comment]
I don't think there is any difference between these two ! _(maybe something I don't know)_

```js
(foo) => this.foo = foo

// VS

(foo) => {
  return this.foo = foo;
}
```

**Comment by ZimGil** [comment]
There's no difference but for the single line return.
In a couple of projects I work on, we do use single line promise then to assign the return value of the promise. 
So IMO we could have an option to allow single line promise then to return the assignment

**Comment by mdjermanovic** [comment]
@ZimGil thanks for the issue!

I'm personally not sure that an additional option would be necessary for this use case, because rule already allows parenthesized assignments:

```js
/* eslint no-return-assign: error */

(foo) => (this.foo = foo) // ok
```

This way an arrow function can return an assignment, the intention to assign value is explicit, and the rule still prevents possible errors where `=` is mistakenly used instead of `==` or `===`.
